doc/source/ray-observability/limiting-memory-usage.rst

```diff
@@ -0,0 +1,296 @@
+Limiting the memory usage
+=========================
+
+The application may consume a large amount of heap space and causes the node to run out of memory. When that happens the operating system will start killing worker or raylet processes, which breaks Ray. It may stall the metrics and if this happens on the head node, it may stall the :ref:`dashboard <ray-dashboard>` or other control processes and causes Ray to fail.
```

Comment:
```suggestion
The application may consume a large amount of heap space and causes the node to run out of memory. When that happens, the operating system will start killing worker or raylet processes which can crash the Ray cluster. It may stall the metrics and if this happens on the head node, it may stall the :ref:`dashboard <ray-dashboard>` or other control processes and causes Ray to fail.
```